mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -457,30 +457,81 @@ def _get_deleted_runs(self):
-            # we assume that the ``latest_metrics`` table already accounts for its presence
-            if just_created:
-                self._update_latest_metric_if_necessary(logged_metric, session)
+            try:
+                run = self._get_run(run_uuid=run_id, session=session)
+                self._check_run_is_active(run)
+
+                metrics_per_key = {}
+
+                for metric in metrics:
+
+                    _validate_metric(metric.key, metric.value, metric.timestamp, metric.step)
+                    is_nan = math.isnan(metric.value)
+                    if is_nan:
+                        value = 0
+                    elif math.isinf(metric.value):
+                        #  NB: Sql can not represent Infs = > We replace +/- Inf
+                        #  with max/min 64b float value
+                        value = 1.7976931348623157e308 if metric.value > 0 \
+                            else -1.7976931348623157e308
+                    else:
+                        value = metric.value
+
+                    # ToDo: Consider prior checks for null, type, metric name validations, ... etc.
+
+                    # Entries will be added to session. If conflicts are detection during saving,
+                    # metrics are inserted one after another using _get_or_create
+                    if auto_new:
+                        logged_metric = SqlMetric(run_uuid=run_id, key=metric.key, value=value,
+                                                  timestamp=metric.timestamp, step=metric.step,
+                                                  is_nan=is_nan)
+                        self._save_to_db(session, logged_metric)
+                        just_created = True
+
+                    else:
+                        logged_metric, just_created = self._get_or_create(
+                            model=SqlMetric, session=session, run_uuid=run_id, key=metric.key,
+                            value=value, timestamp=metric.timestamp, step=metric.step,
+                            is_nan=is_nan)
+
+                    if just_created:
+                        if metric.key not in metrics_per_key:
+                            metrics_per_key[metric.key] = []
+
+                        metrics_per_key[metric.key].append(logged_metric)
+
+                # Conditionally update the ``latest_metrics`` table if the logged metric  was not
+                # already present in the ``metrics`` table. If the logged metric was already
+                # present, we assume that the ``latest_metrics`` table already accounts for
+                # its presence
+
+                for logged_metric_list in metrics_per_key.values():
+                    self._update_latest_metric_if_necessary(None,
+                                                            logged_metrics=logged_metric_list,
+                                                            session=session)
```

Comment:
please put a comment/ToDo that this part is currently not done in batch due to complexity in `_update_latest_metric_if_necessary` step

Reply:
I added a todo